specs/proposal/542678_improve_bundle_image_provided_shape/improve_bundle_image_provided_shape.textile

```diff
@@ -0,0 +1,57 @@
+
+h2. Detailed Specification
+
+Currently, the extension point can modify three properties of an SVG:
+- the color of the shape;
+- the color of the border of the shape;
+- the width of the border of the shape.
+
+For each property, the extension point is used to point the id of the tags and its attribute the manages the property. For example, with the donut.svg sample, the color of the shape is specified in a tag with the id "stop8224-2" and the color is set in an attribute "style". The "style" attribute is in fact a collection of attributes and among them "stop-color" holds the color value (example: _style="stop-color:#ffffff;stop-opacity:1;"_). This "stop-color" sub-attribute is specific to a "LinearGradient" tag and cannot be specified in the extension point. It is currently the attribute that Sirius is expected to find.
+This is a limitation of this extension point as this way to set a color in a tag can be different in other tags. For instance, an ellipse tag also has a "style" attribute and is a collection of sub-attributes (multi-valued attribute). However, this time the color is held by a sub-attributes named "fill" (example: _style="fill:#ffffff;fill-opacity:1;..."_).
+Furthermore, we cannot assume to always have to look for a sub-attributes. The "circle" tag defines its color using a "fill" attribute with no sub-attributes (a single-valued attribute) but directly the color code (example: _fill="#fff"_).
+
+The extension point should be able to specify the sub-attribute that should be edited. If a sub-attribute is not defined in the extension point, then Sirius should consider that this is not a multi-valued attribute (the case of the last example) instead of assuming to look for a specific sub-attribute. 
+
+h2. Backward Compatibility and Migration Paths
+
+A migration will be needed in order to add in the extension point the sub-attribute that were expected to be found such as: _stop-color_, _stroke_ and _stroke-width_.
+
+h3. Metamodel Changes
+
+There is no metamodel change for this feature.
+  
+h3. API Changes
+
+This feature uses the existing API and does not need new API or API changes.
+
+h3. User Interface Changes
+
+The extension point tags _colorAttribute_, _borderColorAttribute_ and __borderSizeAttribute__ will now provide a sub attribute (not mandatory).
+
+h3. Documentation Changes
+
+* Update New and Noteworthy documentation.
+* In developer documentation, the documentation of the "provided bundled image shape extension point":https://www.eclipse.org/sirius/doc/developer/extensions-provide_custom_bundled_image_shape.html needs to be updated.
+
+h2. Tests and Non-regression strategy
+
+We will need to test the color, border color and border size with two new sample:
```

Comment:
a test with the previous version must continue to pass without change (we must be compatible with previous version)

Reply:
Done